.github/workflows/infra-check.yml

```diff
@@ -86,8 +81,14 @@ jobs:
 
           done
 
-          if [ "$failed_check" == "yes" ]; then
-            gh pr comment ${{ github.event.pull_request.number }} -F ./message.txt
+          # Label related actions should only trigger comment if the "infrastructure" label triggered it specifically.
+          skip_comment="no"
+          if ([ "${{ github.event.action }}" == "labeled" ] || [ "${{ github.event.action }}" == "unlabeled" ]) && [ "${{ github.event.label.name }}" != "infrastructure" ]; then
```

Comment:
I wasn't aware there is a way to find out which label was added. This is genius solution!